src/ndarray/ndarray_function.cc

```diff
@@ -207,7 +207,10 @@ void ElementwiseSumContainsDnsImpl(mshadow::Stream<cpu>* s,
   using namespace mxnet::op::mxnet_op;
   const TBlob& out_data = out->data();
   MSHADOW_TYPE_SWITCH(out->dtype(), DType, {  // data type
-    Kernel<set_zero, cpu>::Launch(s, out_data.Size(), out_data.dptr<DType>());
+    // Do not set_zero if output mem inplace with input mem: elemwise_sum.cc FInplaceOption
```

Comment:
Make this comment easy to understand. Add comment that output can be in-placed with the *first* input.